Add mentions field to Reminders model

kosayoda opened the pull request:
To facilitate [bot/#1026](https://github.com/python-discord/bot/issues/1026), this PR adds a field to store the IDs of Member and Role objects.

No validation is done other than ensuring the ID is greater than 0, since we just ignore the ID if it doesn't convert to a valid Member/Role object on the bot side.

kosayoda committed:
Add mentions field to Reminder model

kosayoda committed:
Return mentions from ReminderSerializer

kosayoda committed:
Document mentions in ReminderViewSet

kosayoda committed:
Add migration for the mentions field in the  Reminder model

kosayoda committed:
Document POSTing mentions in ReminderViewSet

kosayoda committed:
Document PATCH for reminders

kosayoda committed:
Document more undocumented stuff

kosayoda committed:
Add mentions field to valid data test

kosayoda committed:
Merge branch 'master' of github.com:python-discord/site into role-reminders

kosayoda committed:
Merge migrations

MarkKoz commented on pydis_site/apps/api/tests/test_reminders.py:
```diff
@@ -54,6 +54,18 @@ def setUpTestData(cls):
+            colour=2,
+            permissions=0b01010010101,
+            position=10,
+        )
```
It isn't necessary to create these objects since there's no FK being used. Just use some integer literals for the IDs in the test.

MarkKoz commented on pydis_site/apps/api/viewsets/bot/reminder.py:
```diff
@@ -43,15 +50,34 @@ class ReminderViewSet(
-    ...     'expiration': str  # ISO-formatted datetime
+    ...     'expiration': str,  # ISO-formatted datetime
+    ...     'channel_id': int,
+    ...     'jump_url': Optional[str]
```
I believe `jump_url` isn't optional according to the model.

MarkKoz reviewed (changes requested):
Is there any compelling reason we'd still need to track the author? Can we just lump their ID in with the rest of the mentions?

kosayoda committed:
Use literal integers for mentions ID in test

Since the mentions field stores static IDs and not foreign keys, there
is no need to create the objects for the test.

kosayoda committed:
Fix misleading documentation

Den4200 reviewed (approved):
It looks good to me.
